spek-ide-plugin/interop-jvm/src/main/kotlin/org/spekframework/ide/ServiceMessageAdapter.kt

```diff
@@ -4,26 +4,32 @@ import org.spekframework.spek2.runtime.execution.ExecutionListener
 import org.spekframework.spek2.runtime.execution.ExecutionResult
 import org.spekframework.spek2.runtime.scope.ActionScopeImpl
 import org.spekframework.spek2.runtime.scope.GroupScopeImpl
+import org.spekframework.spek2.runtime.scope.Path
 import org.spekframework.spek2.runtime.scope.TestScopeImpl
 import java.io.CharArrayWriter
 import java.io.PrintWriter
 
 class ServiceMessageAdapter: ExecutionListener {
-    override fun executionStart() = Unit
+    private val durations = mutableMapOf<Path, Long>()
+    override fun executionStart() { }
 
-    override fun executionFinish() = Unit
+    override fun executionFinish() {
+        durations.clear()
+    }
 
     override fun testExecutionStart(test: TestScopeImpl) {
+        durations[test.path] = System.currentTimeMillis()
         out("testStarted name='${test.path.name.toServiceMessageSafeString()}'")
     }
 
     override fun testExecutionFinish(test: TestScopeImpl, result: ExecutionResult) {
         val name = test.path.name.toServiceMessageSafeString()
+        val duration = System.currentTimeMillis() - (durations[test.path] ?: 0)

```

Comment:
Let's fail if `duration[test.path]` is null? Seems impossible thus no need to write defensive code